### PR TITLE
Move parser test data to an external file

### LINKED ISSUE #651: Move parser/tokenizer test data to an external file
Move test data in `testParser.bal` to an external file due to method too large issue. (jBallerina should fix it under ballerina-platform/ballerina-lang#33534).

### PR DESCRIPTION
**Pull request by manuranga**
Fixes #651

### CODE REVIEW

**Comment by manuranga** [comment]
I decided not to move tokenizer tests for now because it has different structure, it'll make the code ugly, (or we should put that in a separate file).

File: compiler/modules/front.syntax/astString.bal [hunk 1]
```diff
@@ -258,6 +258,12 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
         w.push(CLING, ">");
         return;
     }
+    else if td is ErrorTypeDesc {
+        w.push("map", CLING, "<", CLING);
```
Comment (jclark) [inline compiler/modules/front.syntax/astString.bal]: It's usually a record. Why aren't we calling typeDescToWords on the detail?

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/astString.bal (hunk 1 above)
Comment (manuranga) [reply compiler/modules/front.syntax/astString.bal]: I have made a mistake, this should be `error`.

File: compiler/modules/front.syntax/tests/data/testParser.json [hunk 2]
```diff
@@ -0,0 +1,373 @@
+[
+    ["E", "expr", ""],
+    { "comment" : "not lexical errors" },
```
Comment (jclark) [inline compiler/modules/front.syntax/tests/data/testParser.json:3]: This is a bit ugly: some of these are akin to section names; some apply to a particular test. But I guess it's not worth spending time on at this point.

**Review by jclark — APPROVED** [review]

File: compiler/modules/front.syntax/tests/data/testParser.json (hunk 2 above)
Comment (manuranga) [reply compiler/modules/front.syntax/tests/data/testParser.json:3]: I thought about adding another level, and decided against it thinking everything might not neatly fall into categories.
I'll create an issue to consider in the future.

**Pull request merged**